src/coreComponents/mesh/CellBlock.hpp

```diff
@@ -145,6 +145,20 @@ class CellBlock : public ElementSubRegionBase
                      localIndex_array & nodeIndices ) const;
 
 
+  /**
```

Comment:
This shouldn't be a member. It should ideally be a free function or a member of the finite element/geometry space or something like that. Either way we should absolutely not be capturing a `CellBlock` on device.

Same goes for the `getFaceNodes` above, but you didn't add those.

Reply:
Also since this is templated on the `ELEM_TYPE`, `nodeIndices` could be statically sized. For example for a `ElementType::Hexahedron`, `nodeIndices` could be `localIndex[ 6 ][ 4 ]`. Also you're assuming that they user has properly resized `nodeIndices` prior to this call since no checking is done to make sure it is the correct size.